app/services/vendor_api/generate_test_applications_for_provider.rb

```diff
@@ -0,0 +1,27 @@
+module VendorAPI
+  class GenerateTestApplicationsForProvider
+    def call(provider:, courses_per_application:, count:, for_ratified_courses: false)
+      raise ParameterInvalid, 'Parameter is invalid (cannot be zero): courses_per_application' if courses_per_application.zero?
```

Comment:
After fiddling around refactoring, I think these errors are distinct from the ones inside the TestApplication generator (even if the checks are similar).

Reply:
I think I agree — this is an interface concern and that's a business logic concern